Chessington.GameEngine/ExploreInOneDirection.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Chessington.GameEngine.Tests.Pieces
+{
+    public class ExploreInOneDirection
+    {
+        public static IEnumerable<Square> ExploreOneDirection(Board board, Square initialSquare, int direction)
+        {
+            var squareList = new List<Square>();
+            var nextSquare = iterator(initialSquare, direction);
```

Comment:
this assumes that directions are integers from 0-7, and then you are calling it with specific integers, which makes it unreadable.